torch/csrc/jit/tensorexpr/ir_printer.cpp

```diff
@@ -554,4 +574,40 @@ std::string to_string(const Stmt* stmt) {
   oss << *stmt;
   return oss.str();
 }
+
+std::string to_string(const Tensor* t) {
+  if (!t) {
+    return "(null tensor)\n";
+  }
+  std::ostringstream oss;
+  oss << "Tensor " << t->buf()->name_hint() << "(";
+  for (size_t i = 0; i < t->ndim(); i++) {
+    if (i != 0) {
+      oss << ", ";
+    }
+    oss << *t->arg(i) << "[" << *t->dim(i) << "]";
+  }
+  oss << ") = " << *t->body() << "\n";
+  return oss.str();
+}
+
+std::string to_string(const Function* f) {
```

Comment:
Minor: I mildly prefer using "std::ostream& operator<<" as the canonical way to serialize into text.

Reply:
I added `operator<<` for `Tensor` and `Function` too - do you mean that `to_string` should use those rather than the other way around?